### DISCUSSION TITLE
In-Memory Connector based tests failed in Pulsar.

### DISCUSSION THREAD

**Discussion by hantsy** [quarkusio/quarkus#36764] [Community]
I use an in-memory connector(already set up in [InMemoryProfile](https://github.com/hantsy/quarkus-sandbox/blob/master/pulsar/src/test/java/com/example/InMemoryProfile.java)) to test the internal flow of the message stream without a real message broker. 

The similar codes is working in other connector( amqp, kakfa, rabbitmq, etc.), but failed in Puslar, https://github.com/hantsy/quarkus-sandbox/blob/master/pulsar/src/test/java/com/example/MessageHandlerTest.java

When running this test,I do not need the Pulsar dev service, if there is a way to disable it and clean its related properties?

I am afraid the pulsar properties affected `MessageHandlerTest`, is there some way to print all properties at runtime in the testing codes?

Similar tests in amqp, kakfa, rabbitmq can be found in https://github.com/hantsy/quarkus-sandbox.

**Comment by marcos-lessa** [comment]
You can print all properties using `Config` from `org.eclipse.microprofile.config.Config`

Inject the Config into your test class, then:

```
final Iterable<String> propertyNames = config.getPropertyNames();
		for (final String propertyName : propertyNames) {
			final String propertyValue = config.getValue(propertyName, String.class);
			System.out.println(propertyName + " = " + propertyValue);
		}
```
